File: easy/cpp/0088.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int h1 = m - 1, h2 = n - 1;
        int i = m + n - 1;
        while(h1 > -1 and h2 > -1){
            if(nums1[h1] > nums2[h2]) nums1[i] = nums1[h1--];
            else nums1[i] = nums2[h2--];
            i--;
        }
        for(int j = h1; j > -1; j--) nums1[i--] = nums1[j];
        for(int j = h2; j > -1; j--) nums1[i--] = nums2[j];
    }
};

File: easy/python/0088.py
class Solution:
    def merge(self, nums1: 'List[int]', m: 'int', nums2: 'List[int]', n: 'int') -> 'None':
        """
        Do not return anything, modify nums1 in-place instead.
        """
        h1, h2 = m-1, n-1
        last = m+n-1
        while h1 > -1 and h2 > -1:
            if nums1[h1] > nums2[h2]:
                nums1[last] = nums1[h1]
                h1 -= 1
            else:
                nums1[last] = nums2[h2]
                h2 -= 1
            last -= 1
        for i in range(h1, -1, -1):
            nums1[last] = nums1[i]
            last -= 1
        for i in range(h2, -1, -1):
            nums1[last] = nums2[i]
            last -= 1